### DISCUSSION TITLE
pynetbox - error 401

### DISCUSSION THREAD

**Discussion by y0d4a** [netbox-community/pynetbox#652] [Q&A]
Hi,

I'm trying to add a device in NetBox using Python, but I'm having trouble logging in with the token. I'm using NetBox as integrated within Malcolm.fyi, which is set up behind a reverse proxy. I've tried various configuration changes, but I keep getting a 401 Unauthorized error.

Here's an example of my login check code:
> curl -s -o /dev/null -w "%{http_code}" -H "Authorization: Token XXX" 'https://ids.domain.com/netbox/'

because there is http auth. i tried also with:

> curl -s -o /dev/null -w "%{http_code}" -H "Authorization: Token XXX" 'https://ids.domain.com/netbox/' -u user:pass

I noticed there was an issue with the API in earlier versions. I'm not sure if it's been fixed or if I'm experiencing the same problem...

**Comment by lYlantis** [comment]
Did you ever get the solution to this? Also trying to use Netbox with Malcolm.

**Reply by y0d4a** [reply]
[Keycloak](https://malcolm.fyi/docs/authsetup.html#AuthKeycloak) ?

**Reply by lYlantis** [reply]
My current setup doesn't use Keycloak, so presumably that shouldn't be getting in the way?

**Reply by lYlantis** [reply]
The issue was that I had basic auth enabled, but was trying to use the token. Had to use the user:pass base64 encoded in the request.
